addons/mrp/tests/test_backorder.py

```diff
@@ -356,6 +356,33 @@ def test_backorder_name_without_procurement_group(self):
         self.assertEqual(production.name.split('-')[0], backorder_ids.name.split('-')[0])
         self.assertEqual(int(production.name.split('-')[1]) + 1, int(backorder_ids.name.split('-')[1]))
 
+    def test_split_merge(self):
+        # Change 'Units' rounding to 1 (integer only quantities)
+        self.uom_unit.rounding = 1
+        # Create a mo for 10 products
+        mo = self.generate_mo(qty_final=10)[0]
+        # Split in 3 parts
+        action = mo.action_split_wizard()
+        wizard = Form(self.env[action['res_model']].browse(action['res_id']))
+        wizard.counter = 3
+        action = wizard.save().action_split()
+        # Should have 3 mos
+        self.assertEqual(len(mo.procurement_group_id.mrp_production_ids), 3)
+        mo1 = mo.procurement_group_id.mrp_production_ids[0]
+        mo2 = mo.procurement_group_id.mrp_production_ids[1]
+        mo3 = mo.procurement_group_id.mrp_production_ids[2]
+        # Check quantities
```

Comment:
You should also check the component's quantities to make sure they are correctly split/merged as well.